packages/aws-cdk/bin/cdk.ts

```diff
@@ -347,14 +348,78 @@ async function initCommandLine() {
 
     fs.mkdirpSync(outputDir);
 
+    const extension = json ? 'json' : 'yaml';
+    const hashCache: { [path: string]: string } = {};
     for (const stack of stacks) {
       const finalName = renames.finalName(stack.name);
-      const fileName = `${outputDir}/${finalName}.template.${json ? 'json' : 'yaml'}`;
+      const fileName = path.join(outputDir, `${finalName}.template.${extension}`);
       highlight(fileName);
       await fs.writeFile(fileName, toJsonOrYaml(stack.template));
+      const manifestName = path.join(outputDir, `${finalName}.manifest.${extension}`);
+      await fs.writeFile(manifestName, toJsonOrYaml(await prepareManifest(stack, outputDir, hashCache)));
     }
 
     return undefined; // Nothing to print
+
+    /**
+     * Prepares an asset manifest document, staging assets in a specified directory.
+     * @param stack the stack for which assets are to be manifested.
+     * @param stackName the name of the stack (accounting for renames)
+     * @param dir the directory under which to stage the assets (an assets sub-directory will be used).
+     * @param cache a cache for asset files hashes, improving performance when the same assets are used across multiple stacks
+     */
+    async function prepareManifest(stack: cxapi.SynthesizedStack, dir: string, cache: { [path: string]: string }) {
+      const manifest = { version: '1', assets: {} as any };
+      for (const key of Object.keys(stack.metadata)) {
+        const entries = stack.metadata[key].filter(entry => entry.type === cxapi.ASSET_METADATA);
+        for (const entry of entries) {
+          const asset = entry.data! as cxapi.AssetMetadataEntry;
+          const basePath = path.join('assets', await hash(asset.path), path.basename(asset.path));
+          const fullPath = path.join(dir, basePath);
+          if (!await fs.pathExists(fullPath)) {
+            await fs.copy(asset.path, fullPath);
```

Comment:
Can we use symlinks to make things faster, or would this break CI/CD?

Reply:
The CodeBuild documentation doesn't really state anything with respects to symlinks behavior in output artifacts, so I'd rather avoid assuming they are replaced by the referent file... But on the other hand this should be relatively easy to test out and verify. That'd also save disk space on a user machine, which is not a bad thing.